catroid/src/org/catrobat/catroid/pocketmusic/note/trackgrid/GridRow.java

```diff
@@ -52,7 +51,7 @@ public int hashCode() {
 		int result = 21;
 		result = 31 * result + noteName.hashCode();
```

Comment:
Why 31 is an interesting question. I used this number because that's what's Intellij Ideas default when auto-generating the hashCode function. But i was curious and found [this](https://computinglife.wordpress.com/2008/11/20/why-do-hash-functions-use-prime-numbers/) article, which gives an in-depth explanation.

I do not quite understand what you mean by using a variable for 31? Do you mean a field variable of type static final int? This would then have to be done to all hashCode() functions, which is quite a bit of unnecessary work IMO. But if there is reasonable benefit, i will do this.

Which variable name would be more suitable here? "result" is also auto generated by Intellij Idea. Maybe "hash" or "hashResult"?